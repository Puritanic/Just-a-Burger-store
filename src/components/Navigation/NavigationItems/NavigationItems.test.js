import { shallow, configure } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import NavigationItems from './NavigationItems';
import NavigationItem from './NavigationItem/NavigationItem';

configure({ adapter: new Adapter() });

describe('<NavigationItems />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NavigationItems />);
  });

  it('should render two NavigationItems without crashing', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render two NavigationItems if not auth', () => {
    expect(wrapper.find(NavigationItem)).toHaveLength(2);
  });

  it('should render three NavigationItems if auth', () => {
    wrapper.setProps({ authenticated: true });
    expect(wrapper.find(NavigationItem)).toHaveLength(3);
  });

  it('should render Logout NavItem if auth', () => {
    wrapper.setProps({ authenticated: true });
    expect(wrapper.contains(<NavigationItem link="/logout">Logout</NavigationItem>)).toEqual(true);
  });
});
