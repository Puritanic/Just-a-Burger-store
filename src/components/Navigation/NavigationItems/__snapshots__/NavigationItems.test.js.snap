// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems /> should render two NavigationItems without crashing 1`] = `
<ul>
  <navigationItem
    exact={true}
    link="/"
  >
    Burger Builder
  </navigationItem>
  <navigationItem
    link="/auth"
  >
    Login
  </navigationItem>
</ul>
`;
